File: first/task1.go
package main

import (
	"fmt"
	"math/rand"
	"time"
)

func main() {
	var M, N int
	fmt.Print("Введите количество строк (M): ")
	fmt.Scan(&M)
	fmt.Print("Введите количество столбцов (N): ")
	fmt.Scan(&N)

	if M <= 0 || N <= 0 {
		fmt.Println("Размеры матрицы должны быть положительными числами.")
		return
	}

	matrix := make([][]int, M)
	result := []int{}
	sum := 0

	rand.NewSource(time.Now().UnixNano())

	for i := 0; i < M; i++ {
		matrix[i] = make([]int, N)
		for j := 0; j < N; j++ {
			matrix[i][j] = rand.Intn(101) - 50
		}
	}

	for i := 1; i < M; i += 2 {
		for j := 0; j < N; j++ {
			if j%2 == 0 && matrix[i][j] >= 0 {
				sum += matrix[i][j]
				result = append(result, matrix[i][j])
			}
		}
	}

	fmt.Printf("\nСумма неотрицательных элементов в чётных строках на нечётных местах: %d\n", sum)
	fmt.Print("Новый массив из этих элементов: ")
	for _, val := range result {
		fmt.Printf("%d ", val)
	}
	fmt.Println()
}

File: first/task1.cpp
#include <iostream>
#include <ctime>
#include <vector>

using namespace std;

int main(){
    int M, N;
    cout << "Введите количество строк (M): ";
    cin >> M;
    cout << "Введите количество столбцов (N): ";
    cin >> N;

    if (M <= 0 || N <= 0) {
        cout << "Размеры матрицы должны быть положительными числами.\n";
        return 1;
    }

    vector<vector<int>> matrix(M, vector<int>(N));
    vector<int> result;
    int sum = 0;

    srand(static_cast<unsigned>(time(nullptr)));


    for (int i = 0; i < M; ++i) {
        for (int j = 0; j < N; ++j) {
            matrix[i][j] = rand() % 101 - 50;
            cout << matrix[i][j] << "\t";
        }
        cout << "\n";
    }

    // Поиск и суммирование неотрицательных чисел в чётных строках на нечётных местах
    for (int i = 1; i < M; i += 2) {
        for (int j = 0; j < N; j+=2) {
            if (matrix[i][j] >= 0) {
                sum += matrix[i][j];
                result.push_back(matrix[i][j]);
            }
        }
    }
    cout << "\nСумма неотрицательных элементов в чётных строках на нечётных местах: " << sum << "\n";
    cout << "Новый массив из этих элементов: ";
    for (int val : result) {
        cout << val << " ";
    }
    cout << "\n";

    return 0;
}